src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -188,13 +188,31 @@ private void setUserInfoInThreadContext(User user, Set<String> mappedRoles) {
             joiner.add(String.join(",", user.getRoles()));
             joiner.add(String.join(",", Sets.union(user.getSecurityRoles(), mappedRoles)));
             String requestedTenant = user.getRequestedTenant();
-            if (!Strings.isNullOrEmpty(requestedTenant)) {
-                joiner.add(requestedTenant);
-            }
+            joiner.add(requestedTenant);
+            String tenantAccessToCheck = getTenancyAccess(requestedTenant, mapTenants(user, mappedRoles));
+            joiner.add(tenantAccessToCheck);
+            log.info(joiner);
```

Comment:
do we need this log print?

Reply:
My bad, wrong patch applied. I changed it to debug.